packages/cli/test/unit/util/progress.test.ts

```diff
@@ -53,6 +53,26 @@ describe("progress", () => {
       expect(progress.secondCall.args[0]).to.eql({total, current: total, ratePerSec: 0, percentage: 100});
     });
 
+    it("should call progress with correct values directly reaches to total", () => {
+      const progress = sandbox.spy();
+      const frequencyMs = 50;
+      const total = 8;
+      const needle = showProgress({total, signal: new AbortController().signal, frequencyMs, progress});
```

Comment:
if we only need to abort signal could consider the following to save some resources

```suggestion
      const needle = showProgress({total, signal: new AbortSignal(), frequencyMs, progress});
```